network/hub.go

```diff
@@ -84,22 +88,23 @@ func newHub(mode string, pkey string, database string) *hub {
 
 }
 
-/* sends the message msg to every subscribers of the channel channel */
+// publishOnChannel sends a message to every subscribers of a channel
 func (h *hub) publishOnChannel(msg []byte, channel []byte) {
-
 	var subscribers []int = nil
-	var err error = nil
-	if !bytes.Equal(channel, []byte("/root")) {
-		subscribers, err = db.GetSubscribers(channel)
-		if err != nil {
-			log.Fatal("can't get subscribers", err)
-		}
+	if bytes.Equal(channel, []byte("/root")) {
+		return
+	}
+
+	subscribers = h.actor.GetSubscribers(string(channel))
+	if subscribers == nil {
+		log.Printf("no subscribers to this channel")
+		return
 	}
 
 	for c := range h.connections {
 		//send msgBroadcast to that connection if channel is main channel or is in channel subscribers
 		_, found := lib.Find(subscribers, c.id)
-
+		// && !emptyChannel seems useless
```

Comment:
What does this comment refers to?

Reply:
the fact that if the message is nil the channel should also be nil, but you never know :)